### DISCUSSION TITLE
注册体验很差

### DISCUSSION THREAD

**Discussion by shiny** [geeknote-net/feedback#23] [General]
- 账号名被占用，需要提交注册才能知道
- 在多次重试之后，reCAPTCHA 会变得很反人类

建议：
- 数据埋点，看看进来注册有多少，注册成功多少，费时多少
- blur 时直接检测用户名占用情况
- 直接检测邮箱注册情况
- 检测时需要对请求频率做限制

**Comment by chloerei** [comment]
多谢反馈。之前用户量比较少，以为这个问题不会这么快遇到。会优先处理。

**Reply by shiny** [reply]
甜蜜的烦恼

**Reply by chloerei** [reply]
其实现在也不多，可能常用的被抢占了 😅

**Comment by chloerei** [comment]
- 加了前端校验，值改变的时候检查唯一性
- 修复了一个 recapcha 代码的错误，该错误会导致校验失败。

再次感谢反馈，这个问题可能导致丢失了很多注册用户。有数据埋点工具推荐吗？

https://user-images.githubusercontent.com/49931/183415162-bbef7af8-17fa-40f7-8bb7-1f1e46572701.mov
